saleor/graphql/product/types/attributes.py

```diff
@@ -0,0 +1,103 @@
+    if 'gradient(' in attribute_value:
+        return AttributeValueType.GRADIENT
+    if '://' in attribute_value:
+        return AttributeValueType.URL
+    return AttributeValueType.STRING
+
+
+class AttributeValue(CountableDjangoObjectType):
+    name = graphene.String(description=AttributeValueDescriptions.NAME)
+    slug = graphene.String(description=AttributeValueDescriptions.SLUG)
+    type = AttributeValueType(description=AttributeValueDescriptions.TYPE)
+    value = graphene.String(description=AttributeValueDescriptions.VALUE)
+    translation = graphene.Field(
+        AttributeValueTranslation,
+        language_code=graphene.Argument(
+            LanguageCodeEnum,
+            description='A language code to return the translation for.',
+            required=True),
+        description=(
+            'Returns translated Attribute Value fields '
+            'for the given language code.'),
+        resolver=resolve_translation)
+
+    class Meta:
+        description = 'Represents a value of an attribute.'
+        only_fields = ['id', 'sort_order']
+        interfaces = [relay.Node]
+        model = models.AttributeValue
+
+    def resolve_type(self, *_args):
+        return resolve_attribute_value_type(self.value)
+
+
+class Attribute(CountableDjangoObjectType):
+    name = graphene.String(description=AttributeDescriptions.NAME)
+    slug = graphene.String(description=AttributeDescriptions.SLUG)
+    values = gql_optimizer.field(
+        graphene.List(
+            AttributeValue, description=AttributeDescriptions.VALUES),
+        model_field='values')
+    translation = graphene.Field(
+        AttributeTranslation,
+        language_code=graphene.Argument(
+            LanguageCodeEnum,
+            description='A language code to return the translation for.',
+            required=True),
+        description=(
+            'Returns translated Attribute fields '
+            'for the given language code.'),
+        resolver=resolve_translation)
+
+    class Meta:
+        description = dedent("""
+            Custom attribute of a product. Attributes can be
+            assigned to products and variants at the product type level.""")
+        only_fields = ['id', 'product_type', 'product_variant_type']
+        interfaces = [relay.Node]
+        model = models.Attribute
+
+    def resolve_values(self, *_args):
+        return self.values.all()
+
+
+class SelectedAttribute(graphene.ObjectType):
+    attribute = graphene.Field(
+        Attribute, default_value=None, description=AttributeDescriptions.NAME,
+        required=True)
+    value = graphene.Field(
+        AttributeValue, default_value=None,
+        description='Value of an attribute.', required=True)
+
+    class Meta:
+        description = 'Represents a custom attribute.'
+
+
+class AttributeInput(graphene.InputObjectType):
+    slug = graphene.String(
+        required=True, description=AttributeDescriptions.SLUG)
+    attribute_value = graphene.String(
```

Comment:
I'd change it to "value". I'm only afraid that people using this might pass the display value here instead of slug and wonder why it doesn't work, but I don't have any better idea how to name this field.